### PR TITLE
Correct alerts to work with Cortex single binary

### PR DESCRIPTION
**Pull request by annanay25**
Signed-off-by: Annanay <annanayagarwal@gmail.com>

### CODE REVIEW

File: cortex-mixin/alerts/alerts.libsonnet [hunk 1]
```diff
@@ -448,9 +435,9 @@
           alert: 'CortexGossipMembersMismatch',
           expr: |||
             memberlist_client_cluster_members_count{%s}
-              != on (cluster,namespace) group_left
-            sum(up{job=~".+/(distributor|ingester|querier|cortex|ruler)"}) by (cluster,namespace)
-          ||| % $.namespace_matcher(),
+              != on (%s) group_left
+            sum(up{job=~".+/(distributor|ingester|querier|cortex|ruler)"}) by (%s)
+          ||| % [$.namespace_matcher(), $.aggregation_labels('namespace, job'), $.aggregation_labels('namespace, job')],
```
Comment (jtlisi) [inline cortex-mixin/alerts/alerts.libsonnet:440]:
```suggestion
          ||| % [$.namespace_matcher(), $.aggregation_labels('cluster, namespace'), $.aggregation_labels('cluster, namespace')],
```

File: cortex-mixin/alerts/alerts.libsonnet [hunk 2]
```diff
@@ -295,71 +282,71 @@
...
+              sum by(%s) (cortex_ingester_memory_series * cortex_ingester_chunk_size_bytes_sum / cortex_ingester_chunk_size_bytes_count)
                / 1e9
             )
               >
             (
-              sum by (cluster, namespace) (memcached_limit_bytes{job=~".+/memcached"}) / 1e9
+              sum by(%s) (memcached_limit_bytes{job=~".+/memcached"}) / 1e9
             )
-          |||,
+          ||| % $.aggregation_labels('cluster, namespace'),
```
Comment (jtlisi) [inline cortex-mixin/alerts/alerts.libsonnet:292]:
```suggestion
          ||| % [$.aggregation_labels('cluster, namespace'), $.aggregation_labels('cluster, namespace')],
```

File: cortex-mixin/alerts/alerts.libsonnet [hunk 3]
```diff
@@ -295,71 +282,71 @@
...
           alert: 'CortexProvisioningTooManyActiveSeries',
           // 1 million active series per ingester max.
           expr: |||
-            avg by (cluster, namespace) (cortex_ingester_memory_series) > 1.1e6
+            avg by (%s) (cortex_ingester_memory_series) > 1.1e6
               and
-            sum by (cluster, namespace) (rate(cortex_ingester_received_chunks[1h])) == 0
-          |||,
+            sum by (%s) (rate(cortex_ingester_received_chunks[1h])) == 0
+          ||| % $.aggregation_labels('cluster, namespace'),
```
Comment (jtlisi) [inline cortex-mixin/alerts/alerts.libsonnet:310]:
```suggestion
          ||| % [$.aggregation_labels('cluster, namespace'), $.aggregation_labels('cluster, namespace')],
```

File: cortex-mixin/alerts/alerts.libsonnet [hunk 4]
```diff
@@ -7,36 +7,23 @@
...
+          ||| % [$.namespace_matcher(','), $.aggregation_labels('namespace, job')],
           labels: {
             severity: 'critical',
           },
           annotations: {
             message: '{{ $labels.job }} reports more than one unhealthy ingester.',
           },
         },
-        {
-          alert: 'CortexFlushStuck',
```
Comment (jtlisi) [inline cortex-mixin/alerts/alerts.libsonnet:20]: Removing this seems unrelated to the goal of this PR

Comment (annanay25) [reply cortex-mixin/alerts/alerts.libsonnet:20]: This alert is superseded by `CortexOldChunkInMemory `

File: cortex-mixin/alerts/alerts.libsonnet [hunk 5]
```diff
@@ -295,71 +282,71 @@
...
+              High QPS for ingesters in {{ %s }}, add more ingesters.
+            ||| % $.annotation_labels(),
           },
         },
         {
           alert: 'CortexProvisioningTooMuchMemory',
           expr: |||
-            avg by (cluster, namespace) (container_memory_working_set_bytes{container_name="ingester" %s} / container_spec_memory_limit_bytes{container_name="ingester" %s}) > 0.7
-          ||| % [$.namespace_matcher(','), $.namespace_matcher(',')],
+            avg by (%s) (container_memory_working_set_bytes{container_name="ingester" %s} / container_spec_memory_limit_bytes{container_name="ingester" %s}) > 0.7
```
Comment (annanay25) [inline cortex-mixin/alerts/alerts.libsonnet:340]: This still doesn't work because it uses a container metric `container_memory_working_set_bytes`

Comment (jtlisi) [reply cortex-mixin/alerts/alerts.libsonnet:340]: Since we can't use this metric in all environments we need to either build this instrumentation into Cortex. Or come up with an alert that uses node exporter metrics to accomplish the same task.

**Comment by tomwilkie** [comment]
I think this is overcomplicated - all we need is a string to override the labels we aggregate by in a alert.

I'll make some changes.

**Comment by tomwilkie** [comment]
(Closing in preference for other PR with working CI)
